### PR TITLE
[WIP]Added tableProvider to supply carbonTable wherever needed

### PR DESCRIPTION
**Pull request by dhatchayani**
we can set the carbon table and the carbon table will be supplied through the tableProvider

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3093/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/503/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3095/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/505/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3096/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/506/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3097/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/507/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/508/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3098/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3099/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/509/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3100/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3103/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/512/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3104/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/513/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/514/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3105/

File: core/src/main/java/org/apache/carbondata/core/datastore/TableSegmentUniqueIdentifier.java [hunk 1]
```diff
@@ -34,6 +36,8 @@
    */
   private AbsoluteTableIdentifier absoluteTableIdentifier;
 
+  private TableProvider tableProvider;
```
Comment (gvramana) [inline core/src/main/java/org/apache/carbondata/core/datastore/TableSegmentUniqueIdentifier.java]: Keep table provider out of TableSegmentUniqueIdentifier, logically it does not belong here

File: core/src/main/java/org/apache/carbondata/core/dictionary/generator/IncrementalColumnDictionaryGenerator.java [hunk 2]
```diff
@@ -113,20 +115,21 @@ public IncrementalColumnDictionaryGenerator(CarbonDimension dimension, int maxVa
...
-                    columnIdentifier.getDataType());
+        new DictionaryColumnUniqueIdentifier(tableIdentifier, columnIdentifier,
+            columnIdentifier.getDataType(), tableProvider);
     Boolean isDictExists = CarbonUtil.isFileExistsForGivenColumn(storePath, identifier);
     Dictionary dictionary = null;
     long t1 = System.currentTimeMillis();
     if (isDictExists) {
-      Cache<DictionaryColumnUniqueIdentifier, Dictionary> dictCache = CacheProvider.getInstance()
-              .createCache(CacheType.REVERSE_DICTIONARY, storePath);
+      Cache<DictionaryColumnUniqueIdentifier, Dictionary> dictCache =
```
Comment (gvramana) [inline core/src/main/java/org/apache/carbondata/core/dictionary/generator/IncrementalColumnDictionaryGenerator.java]: Remove all indentation changes

File: core/src/main/java/org/apache/carbondata/core/metadata/AbsoluteTableIdentifier.java [hunk 3]
```diff
@@ -47,13 +48,24 @@
    */
   private CarbonTableIdentifier carbonTableIdentifier;
 
+  private TableProvider tableProvider;
```
Comment (gvramana) [inline core/src/main/java/org/apache/carbondata/core/metadata/AbsoluteTableIdentifier.java]: Locally it should not belong here

**Pull request closed without merging**